SwiftSampleProject/SwiftyVokoderTests/ManagedObjectContextTests.swift

```diff
@@ -0,0 +1,151 @@
+                object: rootContext) { _ in
+                    
+                    let newCountOfStations = self.manager.countForClass(Station.self, forContext: rootContext)
+                    XCTAssertEqual(newCountOfStations, 0)
+                    XCTAssertNotEqual(countOfStations, newCountOfStations)
+                    
+                    return true
+            }
+        }
+        
+        self.manager.saveMainContext()
+        
+        self.waitForExpectationsWithTimeout(10, handler: nil)
+    }
+    
+    func testSaveAndMergeWithMainContextSavesGrandChildren() {
+        let childContext = self.manager.temporaryContext()
+        let grandChildContext = NSManagedObjectContext(concurrencyType: .PrivateQueueConcurrencyType)
+        grandChildContext.parentContext = childContext
+        
+        let countOfStations = self.manager.countForClass(Station.self)
+        XCTAssert(countOfStations > 0)
+        
+        grandChildContext.performBlockAndWait {
+            self.manager.deleteAllObjectsOfClass(Station.self, context: grandChildContext)
+        }
+        
+        self.manager.saveAndMergeWithMainContextAndWait(grandChildContext)
+        
+        let newCountOfStations = self.manager.countForClass(Station.self)
+        XCTAssertEqual(newCountOfStations, 0)
+        XCTAssertNotEqual(countOfStations, newCountOfStations)
+    }
+    
+    func testUnsavedMainContextChangesGetPassedToTempContexts() {
+        let countOfStations = self.manager.countForClass(Station.self)
+        XCTAssert(countOfStations > 0)
+
+        let childContextBeforeChanges = self.manager.temporaryContext()
+        self.manager.deleteAllObjectsOfClass(Station.self, context: nil)
+        let childContextAfterChanges = self.manager.temporaryContext()
+
+        let childCountOfStations = self.manager.countForClass(Station.self, forContext: childContextBeforeChanges)
+        XCTAssertNotEqual(countOfStations, childCountOfStations)
+        XCTAssertEqual(childCountOfStations, 0)
+        
+        let newChildCountOfStations = self.manager.countForClass(Station.self, forContext: childContextAfterChanges)
+        XCTAssertNotEqual(countOfStations, newChildCountOfStations)
+        XCTAssertEqual(newChildCountOfStations, childCountOfStations)
+        XCTAssertEqual(newChildCountOfStations, 0)
+    }
+    
+    func testUnsavedTempContextChangesDoNotGetPassedToMainContext() {
+        let countOfStations = self.manager.countForClass(Station.self)
+        XCTAssert(countOfStations > 0)
+
+        let childContext = self.manager.temporaryContext()
+        self.manager.deleteAllObjectsOfClass(Station.self, context: childContext)
+        
+        let childCountOfStations = self.manager.countForClass(Station.self, forContext: childContext)
+        XCTAssertNotEqual(countOfStations, childCountOfStations)
+        XCTAssertEqual(childCountOfStations, 0)
```

Comment:
Based on the name of this method, I would expect that it would check `self.manager.countForClass(Station.self)` again after deleting everything from the child context, to confirm that the count is **not** zero in the main context.